hosts/galileo/ssi/__init__.py

```diff
@@ -0,0 +1,334 @@
+        asof        Only use SPICE kernels that existed before this date; None
+                    to ignore.
+        gapfill     True to include gapfill CKs. False otherwise.
+        mst_pck     True to include MST PCKs, which update the rotation models
+                    for some of the small moons.
+        irregulars  True to include the irregular satellites;
+                    False otherwise.
+    """
+    SSI.initialize(ck=ck, planets=planets, asof=asof,
+                   spk=spk, gapfill=gapfill,
+                   mst_pck=mst_pck, irregulars=irregulars)
+
+
+#===============================================================================
+class Metadata(object):
+
+    #===========================================================================
+    def __init__(self, label):
+        """Use the label to assemble the image metadata.
+
+        Input:
+            label           The label dictionary.
+
+        Attributes:
+            nlines          A Numpy array containing the data in axis order
+                            (line, sample).
+            nsamples        The time sampling array in (line, sample) axis
+                            order, or None if no time backplane is found in
+                            the file.
+            nframelets
+
+        """
+
+        # Image dimensions
+        self.nlines = label['IMAGE']['LINES']
+        self.nsamples = label['IMAGE']['LINE_SAMPLES']
+
+        # Exposure time
+        exposure_ms = label['EXPOSURE_DURATION']
+        self.exposure = exposure_ms/1000.
+
+        # Filters
+        self.filter = label['FILTER_NAME']
+
+        #TODO: determine whether IMAGE_TIME is the start time or the mid time..
+        self.tstart = julian.tdb_from_tai(
+                        julian.tai_from_iso(label['IMAGE_TIME']))
+        self.tstop = self.tstart + self.exposure
+
+        # Target
+        self.target = label['TARGET_NAME']
+
+        # Telemetry mode
+        self.mode = label['TELEMETRY_FORMAT_ID']
+
+        # Window
+        if 'CUT_OUT_WINDOW' in label:
+            self.window = label['CUT_OUT_WINDOW']
+        else:
+            self.window = None
+
+    #===========================================================================
+    def trim(self, data, full_fov=False):
+        """Trim image to label window
+
+        Input:
+            full_fov        If True, the image is not trimmed.
+
+        Attributes:
+            nlines          A Numpy array containing the data in axis order
+                            (line, sample).
+            nsamples        The time sampling array in (line, sample) axis
+                            order, or None if no time backplane is found in
+                            the file.
+            nframelets
+
+        """
+
+        if full_fov:
+            return None
+
+        window = self.window
+
+        if window is None:
+            return data
+
+        return data[window[0]:window[2], window[1]:window[3]]
```

Comment:
I'm not sure of the Galileo convention, but are the window dimensions non-inclusive of the upper limit?

Reply:
It is, but this code is still incorrect.  First, it's (line,sample), not (u,v); second, the second pair of numbers is the size of the region, not the corner.  All of the test images used so far start at (1,1), so the second pair of numbers is both the corner coordniate and the size of the region.  I just looked at some others and confirmed this.

Reply:
I added a gold-master image with a cutout window far from the origin and it looks like this is correct now.